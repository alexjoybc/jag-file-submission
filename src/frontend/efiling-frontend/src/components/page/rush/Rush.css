.form-parent {
  display: flex;
  justify-content: space-between;
  padding-bottom: 15px;
}

.form-child {
  width: 45%;
}

.dropzone-outer {
  border: 1px dashed #707070;
  height: 200px;
  display: flex;
  justify-content: center;
}
