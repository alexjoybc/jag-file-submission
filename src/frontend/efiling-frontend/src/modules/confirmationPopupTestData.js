/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn"
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width"
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width"
};

const handleClose = () => {};
const handleShow = () => {};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: true,
    handleShow,
    handleClose,
    handleConfirm: () => {},
    title: "title",
    body
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
