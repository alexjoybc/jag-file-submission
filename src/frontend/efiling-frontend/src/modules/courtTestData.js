const court = {
  locationDescription: "Court location",
  fileNumber: "Court file number",
  levelDescription: "Level",
  classDescription: "Class",
  level: "P",
  courtClass: "F",
};

export function getCourtData() {
  return court;
}
