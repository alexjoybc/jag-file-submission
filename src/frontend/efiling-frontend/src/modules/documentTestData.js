const documents = [
  {
    name: "file name 1",
    description: "file description 1",
    type: "file type",
    statutoryFeeAmount: 40
  },
  {
    name: "file name 2",
    description: "file description 2",
    type: "file type",
    statutoryFeeAmount: 0
  }
];

export function getDocumentsData() {
  return documents;
}
