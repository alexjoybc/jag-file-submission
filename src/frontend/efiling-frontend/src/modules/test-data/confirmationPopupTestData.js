/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const onButtonClick = () => {};

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn",
  onClick: onButtonClick,
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width",
  onClick: onButtonClick,
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width",
  onClick: onButtonClick,
};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: false,
    title: "title",
    body,
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
