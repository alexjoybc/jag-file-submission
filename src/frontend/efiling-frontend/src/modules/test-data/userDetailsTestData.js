const userDetails = {
  internalClientNumber: "ABC123",
  accounts: [
    {
      type: "CSO",
      identifier: "123",
    },
  ],
};

export function getUserDetails() {
  return userDetails;
}
