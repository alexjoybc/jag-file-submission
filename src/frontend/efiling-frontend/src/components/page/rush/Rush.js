import React, { useState } from "react";
import Dropzone from "react-dropzone";
import { Radio, Callout, Button, Input, Textarea } from "shared-components";

import "./Rush.css";

const calloutText = `Please provide the date of when the direction was made,
the name of the Judge who made the direction along with any additional details
you feel are necessary.`;

const generateInputField = (input, onChange) => {
  return (
    <div className="form-child">
      <Input input={input} onChange={onChange} />
    </div>
  );
};

export default function Rush() {
  const [rushReason, setRushReason] = useState(null);
  const [showPayment, setShowPayment] = useState(false);
  const [acceptedFiles, setAcceptedFiles] = useState([]);
  const input = {
    isReadOnly: false,
    styling: "editable-white",
    isRequired: true,
  };

  return (
    <div className="page">
      <div className="content col-md-8">
        <h1>Request Rush/Urgent Submission</h1>
        <p>
          You must provide a reason and contact information for your request for
          rush/urgent submission to be considered. My reason for requesting
          urgent (rush) filing is:
        </p>
        <Radio
          id="rule-85"
          name="rush-reason"
          label="The attached application is made under Rule 8-5 (1) SCR."
          onSelect={(val) => setRushReason(val)}
        />
        <Radio
          id="court-directed"
          name="rush-reason"
          label="The court directed that the order be processed on an urgent basis."
          onSelect={(val) => setRushReason(val)}
        />
        <Radio
          id="other"
          name="rush-reason"
          label="Other (please explain)."
          onSelect={(val) => setRushReason(val)}
        />
        <Callout text={calloutText} />
        <Textarea
          id="detailed-reason"
          label="Clear and detailed reason(s)"
          onChange={(val) => console.log(val)}
        />
        <p>
          I understand the registry has the discretion to reject this request.
        </p>
        <div className="form-parent">
          {generateInputField(
            {
              ...input,
              label: "Last Name",
              id: "lastName",
            },
            () => console.log("lastname")
          )}
          {generateInputField(
            {
              ...input,
              label: "Organization",
              id: "org",
              isRequired: false,
            },
            () => console.log("org")
          )}
        </div>
        <div className="form-parent">
          {generateInputField(
            {
              ...input,
              label: "First Name",
              id: "firstName",
            },
            () => console.log("firstname")
          )}
          {generateInputField(
            {
              ...input,
              label: "Contact Country",
              id: "country",
              isRequired: false,
            },
            () => console.log("country")
          )}
        </div>
        <div className="form-parent">
          {generateInputField(
            {
              ...input,
              label: "Phone Number",
              id: "phoneNumber",
            },
            () => console.log("phone")
          )}
        </div>
        <br />
        <Dropzone
          onDrop={(droppedFiles) =>
            setAcceptedFiles(acceptedFiles.concat(droppedFiles))
          }
        >
          {({ getRootProps, getInputProps }) => (
            <div
              data-testid="dropdownzone"
              {...getRootProps({ className: "dropzone-outer" })}
            >
              <div className="dropzone-inner-box">
                <input {...getInputProps()} />
                <span>
                  <h2 className="text-center-alignment">
                    Drag and drop or&nbsp;
                    <span className="file-href">choose documents</span>
                  </h2>
                  <div>
                    <span>
                      to upload supporting documents from counsel that explain
                      the reasons for an urgent/rush order.
                    </span>
                    <br />
                    <span>
                      Please note that documents uploaded on this screen will
                      not be processed into your eFile.
                    </span>
                  </div>
                </span>
              </div>
            </div>
          )}
        </Dropzone>
        <br />
        <section className="buttons pt-2">
          <Button
            label="Cancel Request"
            onClick={() => setShowPayment(true)}
            styling="normal-white btn"
          />
          <Button
            label="Submit Request"
            onClick={() => console.log("submit rush")}
            styling="normal-blue btn"
          />
        </section>
      </div>
    </div>
  );
}
