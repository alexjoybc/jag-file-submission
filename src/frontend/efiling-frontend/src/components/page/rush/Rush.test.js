// TODO: remove ignoring Rush.js from coverage report
import React from "react";
import { render } from "@testing-library/react";
import { getDocumentsData } from "../../../modules/test-data/documentTestData";
import { getTestData } from "../../../modules/test-data/confirmationPopupTestData";
import { getCourtData } from "../../../modules/test-data/courtTestData";

import Rush from "./Rush";

describe("Rush Component", () => {
  const confirmationPopup = getTestData();
  const submissionId = "abc123";
  const courtData = getCourtData();
  const files = getDocumentsData();
  const submissionFee = 25.5;

  const payment = {
    confirmationPopup,
    submissionId,
    courtData,
    files,
    submissionFee,
  };

  test("Matches the snapshot", () => {
    const { asFragment } = render(<Rush payment={payment} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
