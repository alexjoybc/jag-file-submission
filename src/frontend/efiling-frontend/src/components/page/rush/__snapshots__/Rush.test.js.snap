// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rush Component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="page"
  >
    <div
      class="content col-md-8"
    >
      <h1>
        Request Rush/Urgent Submission
      </h1>
      <p>
        You must provide a reason and contact information for your request for rush/urgent submission to be considered. My reason for requesting urgent (rush) filing is:
      </p>
      <label
        class="bcgov-radio"
        for="rule-85"
      >
        The attached application is made under Rule 8-5 (1) SCR.
        <input
          id="rule-85"
          name="rush-reason"
          type="radio"
        />
        <span
          class="bcgov-dot"
        />
      </label>
      <label
        class="bcgov-radio"
        for="court-directed"
      >
        The court directed that the order be processed on an urgent basis.
        <input
          id="court-directed"
          name="rush-reason"
          type="radio"
        />
        <span
          class="bcgov-dot"
        />
      </label>
      <label
        class="bcgov-radio"
        for="other"
      >
        Other (please explain).
        <input
          id="other"
          name="rush-reason"
          type="radio"
        />
        <span
          class="bcgov-dot"
        />
      </label>
      <div
        class="bcgov-callout"
      >
        <p>
          Please provide the date of when the direction was made,
the name of the Judge who made the direction along with any additional details
you feel are necessary.
        </p>
      </div>
      <div
        class="input-label"
      >
        <label
          for="Date"
        >
          Date
        </label>
        <span
          class="mandatory"
          id="asterisk"
        >
          *
        </span>
         
      </div>
      <div
        class="bcgov-flex"
      >
        <svg
          class="bcgov-move-right"
          fill="currentColor"
          height="32"
          size="32"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
          />
        </svg>
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class=""
              type="text"
              value="08/13/2020"
            />
          </div>
        </div>
      </div>
      <br />
      <div
        class="input-label"
      >
        <label
          for="detailed-reason"
        >
          Clear and detailed reason(s)
        </label>
        <span
          class="mandatory"
          id="asterisk"
        >
          *
        </span>
         
      </div>
      <textarea
        class="text-input"
        cols="60"
        id="detailed-reason"
        rows="8"
      />
      <p>
        I understand the registry has the discretion to reject this request.
      </p>
      <div
        class="form-parent"
      >
        <div
          class="form-child"
        >
          <div>
            <label
              class="label"
              for="lastName"
            >
              <div
                class="input-label"
              >
                Last Name
              </div>
              <span
                class="mandatory"
                id="asterisk"
              >
                *
              </span>
               
              <span
                class="note"
              />
            </label>
            <input
              class="editable-white"
              id="lastName"
              type="text"
              value=""
            />
            <br />
            <span
              class="error"
            />
          </div>
        </div>
        <div
          class="form-child"
        >
          <div>
            <label
              class="label"
              for="org"
            >
              <div
                class="input-label"
              >
                Organization
              </div>
               
              <span
                class="note"
              />
            </label>
            <input
              class="editable-white"
              id="org"
              type="text"
              value=""
            />
            <br />
            <span
              class="error"
            />
          </div>
        </div>
      </div>
      <div
        class="form-parent"
      >
        <div
          class="form-child"
        >
          <div>
            <label
              class="label"
              for="firstName"
            >
              <div
                class="input-label"
              >
                First Name
              </div>
              <span
                class="mandatory"
                id="asterisk"
              >
                *
              </span>
               
              <span
                class="note"
              />
            </label>
            <input
              class="editable-white"
              id="firstName"
              type="text"
              value=""
            />
            <br />
            <span
              class="error"
            />
          </div>
        </div>
        <div
          class="form-child"
        >
          <div>
            <label
              class="label"
              for="country"
            >
              <div
                class="input-label"
              >
                Contact Country
              </div>
               
              <span
                class="note"
              />
            </label>
            <input
              class="editable-white"
              id="country"
              type="text"
              value=""
            />
            <br />
            <span
              class="error"
            />
          </div>
        </div>
      </div>
      <div
        class="form-parent"
      >
        <div
          class="form-child"
        >
          <div>
            <label
              class="label"
              for="phoneNumber"
            >
              <div
                class="input-label"
              >
                Phone Number
              </div>
              <span
                class="mandatory"
                id="asterisk"
              >
                *
              </span>
               
              <span
                class="note"
              />
            </label>
            <input
              class="editable-white"
              id="phoneNumber"
              type="text"
              value=""
            />
            <br />
            <span
              class="error"
            />
          </div>
        </div>
      </div>
      <br />
      <div
        class="dropzone-outer"
        data-testid="dropdownzone"
        tabindex="0"
      >
        <div
          class="dropzone-inner-box"
        >
          <input
            autocomplete="off"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <span>
            <h2
              class="text-center-alignment"
            >
              Drag and drop or 
              <span
                class="file-href"
              >
                choose documents
              </span>
            </h2>
            <div>
              <span>
                to upload supporting documents from counsel that explain the reasons for an urgent/rush order.
              </span>
              <br />
              <span>
                Please note that documents uploaded on this screen will not be processed into your eFile.
              </span>
            </div>
          </span>
        </div>
      </div>
      <br />
      <section
        class="buttons pt-2"
      >
        <button
          class="bcgov-button normal-white btn"
          data-test-id=""
          type="button"
        >
          Cancel Request
        </button>
        <button
          class="bcgov-button normal-blue btn"
          data-test-id=""
          type="button"
        >
          Submit Request
        </button>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
